modules/runes/runes.go

```diff
@@ -0,0 +1,89 @@
+package runes
+
+import (
+	"context"
+	"strings"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/cockroachdb/errors"
+	"github.com/gaze-network/indexer-network/common/errs"
+	"github.com/gaze-network/indexer-network/core/datasources"
+	"github.com/gaze-network/indexer-network/core/indexer"
+	"github.com/gaze-network/indexer-network/core/types"
+	"github.com/gaze-network/indexer-network/internal/config"
+	"github.com/gaze-network/indexer-network/internal/postgres"
+	runesapi "github.com/gaze-network/indexer-network/modules/runes/api"
+	runesdatagateway "github.com/gaze-network/indexer-network/modules/runes/datagateway"
+	runespostgres "github.com/gaze-network/indexer-network/modules/runes/repository/postgres"
+	runesusecase "github.com/gaze-network/indexer-network/modules/runes/usecase"
+	"github.com/gaze-network/indexer-network/pkg/btcclient"
+	"github.com/gaze-network/indexer-network/pkg/logger"
+	"github.com/gaze-network/indexer-network/pkg/reportingclient"
+	"github.com/gofiber/fiber/v2"
+	"github.com/samber/do/v2"
+	"github.com/samber/lo"
+)
+
+func New(injector do.Injector) (indexer.IndexerWorker, error) {
+	ctx := do.MustInvoke[context.Context](injector)
+	conf := do.MustInvoke[config.Config](injector)
+	btcClient := do.MustInvoke[*rpcclient.Client](injector)
```

Comment:
btcClient should be invoked inside switch case below, as its usage depends on config